src/bundle/ui-dev/src/modules/universal-discovery/search.tab.module.js

```diff
@@ -1,20 +1,30 @@
-import React, { useContext } from 'react';
+import React, { useContext, useEffect } from 'react';
 
 import Tab from './components/tab/tab';
 import Search from './components/search/search';
 
-import { TabsConfigContext } from './universal.discovery.module';
+import { LoadedLocationsMapContext, MarkedLocationIdContext, TabsConfigContext } from './universal.discovery.module';
 
 const { ibexa, Translator } = window;
 
 const SearchTabModule = () => {
     const tabsConfig = useContext(TabsConfigContext);
+    const [markedLocationId, setMarkedLocationId] = useContext(MarkedLocationIdContext);
+    const [loadedLocationsMap, dispatchLoadedLocationsAction] = useContext(LoadedLocationsMapContext);
+
     const actionsDisabledMap = {
         'content-create-button': false,
         'sort-switcher': true,
         'view-switcher': true,
     };
 
+    useEffect(() => {
+        return () => {
+            setMarkedLocationId(markedLocationId);
+            dispatchLoadedLocationsAction({ type: 'SET_LOCATIONS', data: loadedLocationsMap });
```

Comment:
Setting the same values seems a little bit hacky, can you explain why this should be done like that and why browser tab is empty? What is missing there?

Reply:
@dew326 The browse tab is empty because after using the 'Search' module the last action for `LoadedLocationActions` is `CLEAR_LOCATIONS`. Therefore Locations in the browse tab are empty.

We are doing pretty similar thing in the Bookmarks module - https://github.com/ibexa/admin-ui/blob/main/src/bundle/ui-dev/src/modules/universal-discovery/bookmarks.tab.module.js#L54.